internal/provider/data_source_boundary_cluster.go

```diff
@@ -44,6 +44,34 @@ func dataSourceBoundaryCluster() *schema.Resource {
 				Type:        schema.TypeString,
 				Computed:    true,
 			},
+			"maintenance_window_config": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"upgrade_type": {
+							Description: "The upgrade type for the cluster",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"maintenance_window_day": {
+							Description: "The day of the week for scheduled updates",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"maintenance_window_start": {
+							Description: "The start time frame for the maintenance window",
+							Type:        schema.TypeInt,
+							Computed:    true,
+						},
+						"maintenance_window_end": {
+							Description: "The end time frame for the maintenance window",
+							Type:        schema.TypeInt,
+							Computed:    true,
+						},
```

Comment:
Personally I think the `maintenance_window` prefix here is a bit extraneous, it will already be in the maintenance window block so just `start` and `end` would be less typing.